docs/merkletree/treetex/main.go

```diff
@@ -321,14 +327,21 @@ func modifyRangeNodeInfo() error {
 	return nil
 }
 
+var dataFormat = func(id compact.NodeID) string {
+	return fmt.Sprintf("{$leaf_{%d}$}", id.Index)
+}
+
 var nodeFormats = map[string]nodeTextFunc{
 	"address": func(id compact.NodeID) string {
 		return fmt.Sprintf("%d.%d", id.Level, id.Index)
 	},
 	"hash": func(id compact.NodeID) string {
-		childLevel := id.Level - 1
-		leftChild := id.Index * 2
-		return fmt.Sprintf("{$H_{%d.%d} =$ \\\\ $H(H_{%d.%d} || H_{%d.%d})$}", id.Level, id.Index, childLevel, leftChild, childLevel, leftChild+1)
+		if id.Level >= 1 {
+			childLevel := id.Level - 1
+			leftChild := id.Index * 2
+			return fmt.Sprintf("{$H_{%d.%d} =$ \\\\ $H(H_{%d.%d} || H_{%d.%d})$}", id.Level, id.Index, childLevel, leftChild, childLevel, leftChild+1)
+		}
+		return fmt.Sprintf("{$H_{%d.%d} =$ \\\\ $H(leaf_{%[2]d})$}", id.Level, id.Index)
```

Comment:
What does `%[2]d` do? Can't find it in documentation.

Reply:
This makes the fmt specifier take from the `nth` (1-based index) parameter, so in this case it's saying _"re-use the 2nd parameter here (even though it was 'consumed' earlier in the format string)"_

Reply:
Wow, good to know.

Can't say I like this feature too much: "After processing a bracketed expression [n], subsequent verbs will use arguments n+1, n+2, etc. unless otherwise directed."